src/metabase/pulse/render/common.clj

```diff
@@ -25,55 +28,112 @@
   Object
   (toString [_] num-str))
 
+(defn- strip-trailing-zeroes
+  [num-as-string decimal]
+  (if (str/includes? num-as-string (str decimal))
+    (let [pattern (re-pattern (str/escape (str decimal \$) {\. "\\."}))]
+      (-> num-as-string
+          (str/split #"0+$")
+          first
+          (str/split pattern)
+          first))
+    num-as-string))
+
+(defn- digits-after-decimal
+  ([value] (digits-after-decimal value "."))
+  ([value decimal]
+   (if (zero? value)
+     0
+     (let [val-string (-> (condp = (type value)
+                            java.math.BigDecimal (.toPlainString ^BigDecimal value)
+                            java.lang.Double (format "%.20f" value)
+                            java.lang.Float (format "%.20f" value)
+                            (str value))
+                          (strip-trailing-zeroes (str decimal)))
+           [_n d] (str/split val-string #"[^\d*]")]
+       (count d)))))
+
+(defn- sig-figs-after-decimal
+  [value decimal]
+  (if (zero? value)
+    0
+    (let [val-string (-> (condp = (type value)
+                           java.math.BigDecimal (.toPlainString ^BigDecimal value)
+                           java.lang.Double (format "%.20f" value)
+                           java.lang.Float (format "%.20f" value)
+                           (str value))
+                         (strip-trailing-zeroes (str decimal)))
+          figs (last (str/split val-string #"[\.0]+"))]
+      (count figs))))
+
 (defn number-formatter
   "Return a function that will take a number and format it according to its column viz settings. Useful to compute the
   format string once and then apply it over many values."
-  [{:keys [effective_type base_type] col-id :id col-name :name  :as _column} viz-settings]
-  (let [column-settings    (or (get-in viz-settings [::mb.viz/column-settings {::mb.viz/field-id col-id}])
-                               (get-in viz-settings [::mb.viz/column-settings {::mb.viz/column-name col-name}]))
-        global-settings    (::mb.viz/global-column-settings viz-settings)
-        currency?          (boolean (or (= (::mb.viz/number-style column-settings) "currency")
-                                        (and (nil? (::mb.viz/number-style column-settings))
-                                             (or
-                                              (::mb.viz/currency-style column-settings)
-                                              (::mb.viz/currency column-settings)))))
-
+  [{:keys [effective_type base_type] col-id :id field-ref :field_ref col-name :name :as _column} viz-settings]
+  (let [col-id (or col-id (second field-ref))
```

Comment:
Nit: `let` indentation

Reply:
Hmmm... not sure, I think it's fine?

Reply:
Oh, I suppose you mean the s-exprs, not the symbols in the let?